compiler/rustc_ty_utils/src/ty.rs

```diff
@@ -243,37 +243,41 @@ fn param_env_reveal_all_normalized(tcx: TyCtxt<'_>, def_id: DefId) -> ty::ParamE
     tcx.param_env(def_id).with_reveal_all_normalized(tcx)
 }
 
-/// If `def_id` is an issue 33140 hack impl, returns its self type; otherwise, returns `None`.
+/// If the given trait impl enables exploiting the former order dependence of trait objects,
+/// returns its self type; otherwise, returns `None`.
 ///
-/// See [`ty::ImplOverlapKind::Issue33140`] for more details.
-fn issue33140_self_ty(tcx: TyCtxt<'_>, def_id: DefId) -> Option<EarlyBinder<Ty<'_>>> {
-    debug!("issue33140_self_ty({:?})", def_id);
+/// See [`ty::ImplOverlapKind::FutureCompatOrderDepTraitObjects`] for more details.
+#[instrument(level = "debug", skip_all)]
+fn self_ty_of_trait_impl_enabling_order_dep_trait_object_exploit(
+    tcx: TyCtxt<'_>,
+    def_id: DefId,
+) -> Option<EarlyBinder<Ty<'_>>> {
+    debug!(?def_id);
```

Comment:
```suggestion
#[instrument(level = "debug", skip(tcx))]
fn self_ty_of_trait_impl_enabling_order_dep_trait_object_exploit(
    tcx: TyCtxt<'_>,
    def_id: DefId,
) -> Option<EarlyBinder<Ty<'_>>> {
```
was there a reason you didn't do this?

Reply:
Yeah, I did that intentionally. There's a minor difference. If I just add it to the `tracing` span via `#[instrument]`, the `DefId` doesn't get logged right away but only together with each `debug!` (and for context it wouldn't get logged at all if we didn't reach any `debug!` which won't be the case here).

I'm gonna check the actual output later since my knowledge is based on experiments I did in the past with the default `tracing_subscriber` setup and see how it actually fares in practice.

Reply:
Output with `#[instrument(skip_all)]` & `debug!(?def_id)`:

```
│ │ │ ├─┐rustc_ty_utils::ty::self_ty_of_trait_impl_enabling_order_dep_trait_object_hack 
│ │ │ │ ├─  0ms DEBUG rustc_ty_utils::ty def_id=DefId(0:5 ~ lint_incoherent_auto_trait_objects[0a81]::{impl#1})
│ │ │ │ ├─  0ms DEBUG rustc_ty_utils::ty trait_ref=<(dyn std::marker::Send + 'static) as Foo>
│ │ │ │ ├─  0ms DEBUG rustc_ty_utils::ty MATCHES!
│ │ │ ├─┘
```

Output with `#[instrument(skip(tcx)]` & no debug call:

```
│ │ │ ├─┐rustc_ty_utils::ty::self_ty_of_trait_impl_enabling_order_dep_trait_object_hack def_id=DefId(0:5 ~ lint_incoherent_auto_trait_objects[0a81]::{impl#1})
│ │ │ │ ├─  0ms DEBUG rustc_ty_utils::ty trait_ref=<(dyn std::marker::Send + 'static) as Foo>
│ │ │ │ ├─  0ms DEBUG rustc_ty_utils::ty MATCHES!
│ │ │ ├─┘
```

Yeah, okay, in this case it doesn't matter. Taking your suggested approach.